models/src/main/protobuf/CasperMessage.proto

```diff
@@ -17,6 +17,7 @@ option (scalapb.options) = {
   flat_package: true
   single_file: true
   preamble: "sealed trait CasperMessageProto"
+  preserve_unknown_fields: false
```

Comment:
What can we gain from this option?
I read https://scalapb.github.io/docs/customizations/.
So when the deserialization happened and there is unknown fields, would it throw an error?

Reply:
> By default, during deserialization only known fields are retained. When setting preserve_unknown_fields to true, all generated messages in this file will preserve unknown fields. This is default behaviour in java for Proto3 messages since 3.5.0. In ScalaPB 0.10.0: the default of this field became true for consistency with Java.
https://scalapb.github.io/docs/customizations/

This is now necessary because default behavior changed in Proto 3 release. When is set to _true_ it generates additional parameter in constructor for unknown fields. 

Previously in [ProtoM](https://github.com/rchain/rchain/pull/3197/files#diff-333dc227c0ebc66e94186b2966486e999d6de69db642ff824ef94194875ef129L32-L35) there was a code to throw exceptions if `preserve_unknown_fields` is set, but scalapb extension method `descriptor.preservesUnknownFields` is now not available (needs proto file to construct the implicits). But I think this should not be a problem because it's should always return _false_.

We are depending too much on scalapb and protobuf default behavior on which we don't have any influence. Maybe this is good enough for non-blockchain projects but we should have very precise control how objects are serialized and with what defaults. After all protobuf is just a serializer of primitive types that can be combined in sequence.